src/olympia/addons/views.py

```diff
@@ -974,18 +975,60 @@ class ReplacementAddonView(ListAPIView):
 
 
 class CompatOverrideView(ListAPIView):
+    """
+    This view is being used by Firefox and thus is performance-critical.
+    """
     queryset = CompatOverride.objects.all()
     serializer_class = CompatOverrideSerializer
 
-    def filter_queryset(self, queryset):
+    @classmethod
+    def as_view(cls, **initkwargs):
+        """The API is read-only so we can turn off atomic requests."""
+        return non_atomic_requests(
+            super(CompatOverrideView, cls).as_view(**initkwargs))
+
+    def get_guids(self):
         # Use the same Filter we use for AddonSearchView for consistency.
         guid_filter = AddonGuidQueryParam(self.request)
-        guids = guid_filter.get_value()
+        return guid_filter.get_value()
+
+    def list(self, request, *args, **kwargs):
+        """Override the `list` logic to integrate custom caching."""
+        guids = self.get_guids()
         if not guids:
             raise exceptions.ParseError(
                 'Empty, or no, guid parameter provided.')
-        return queryset.filter(guid__in=guids).transform(
-            CompatOverride.transformer).order_by('-pk')
+
+        _key = CompatOverride.get_api_cache_key
+        results = cache.get_many([_key(guid) for guid in guids]).values()
+        missing = set(guids) - set(item['addon_guid'] for item in results)
+
+        if missing:
+            # Evaluate the actual queryset. The amount of GUIDs we should get
+            # in real-life won't be paginated most of the time so it's safe to
+            # simply evaluate the query. The advantage here is that we are
+            # saving ourselves a `COUNT` query and these are expensive.
+            qset = list(
+                self.get_queryset().filter(guid__in=missing)
+                .transform(CompatOverride.transformer)
+                # We are ordering manually because we have to merge cache +
+                # database fetched entries
+                .order_by())
+
+            if qset:
+                missing_data = self.get_serializer(qset, many=True).data
+                results.extend(missing_data)
```

Comment:
So, I think the flaw here (if I've not missed something) is most guids won't have matches, so they won't be in the queryset response, so they won't be cached, and we'll have to query for them in the database every time.  Although if we save the misses then we'll likely be filling up memcache with a lot of guids.

Reply:
aaaand, I commented after you just said the same thing.

Reply:
> Although if we save the misses then we'll likely be filling up memcache with a lot of guids.

Yeah, that may be problematic but shouldn't be too worrisome. Although, I'm interested in how many overrides there are in total. If the number is reasonably low we can just avoid calling `memcached` because then MySQL should have the whole table in the buffer somehow anyway.